drivers/spi/spi_silabs_eusart.c

```diff
@@ -298,21 +825,43 @@ static DEVICE_API(spi, spi_silabs_eusart_api) = {
 	.release = spi_silabs_eusart_release,
 };
 
-#define SPI_INIT(n)										\
-	PINCTRL_DT_INST_DEFINE(n);								\
-	static struct spi_silabs_eusart_data spi_silabs_eusart_data_##n = {			\
-		SPI_CONTEXT_INIT_LOCK(spi_silabs_eusart_data_##n, ctx),				\
-		SPI_CONTEXT_INIT_SYNC(spi_silabs_eusart_data_##n, ctx),				\
-		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(n), ctx)};				\
-	static struct spi_silabs_eusart_config spi_silabs_eusart_cfg_##n = {			\
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),					\
-		.base = (EUSART_TypeDef *)DT_INST_REG_ADDR(n),					\
-		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),				\
-		.clock_cfg = SILABS_DT_INST_CLOCK_CFG(n),					\
-		.clock_frequency = DT_INST_PROP_OR(n, clock_frequency, 1000000)			\
-	};											\
-	SPI_DEVICE_DT_INST_DEFINE(n, spi_silabs_eusart_init, NULL, &spi_silabs_eusart_data_##n,	\
-			      &spi_silabs_eusart_cfg_##n, POST_KERNEL, CONFIG_SPI_INIT_PRIORITY,\
-			      &spi_silabs_eusart_api);
+#if defined(CONFIG_SPI_SILABS_EUSART_DMA)
+#define DMA_ENABLED_INSTANCE(n) \
+	.is_dma_enabled_instance = COND_CODE_1(DT_INST_NODE_HAS_PROP(n, dmas), (1), (0U)),
+
+#define SPI_DMA_CHANNEL_INIT(index, dir) \
+	.dma_channel_##dir = { \
+		.dma_device = DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(index, dir)), \
+		.dma_slot = SILABS_LDMA_REQSEL_TO_SLOT(DT_INST_DMAS_CELL_BY_NAME(index, dir, slot)), \
+	},
+#define SPI_DMA_CHANNEL(index, dir) \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(index, dmas), (SPI_DMA_CHANNEL_INIT(index, dir)), ())
+#else
+#define SPI_DMA_CHANNEL(index, dir)
+#define DMA_ENABLED_INSTANCE(n)
+#endif
+
+#define SPI_INIT(n) \
+	PINCTRL_DT_INST_DEFINE(n); \
+	static struct spi_silabs_eusart_data spi_silabs_eusart_data_##n = { \
+		SPI_CONTEXT_INIT_LOCK(spi_silabs_eusart_data_##n, ctx), \
+		SPI_CONTEXT_INIT_SYNC(spi_silabs_eusart_data_##n, ctx), \
+		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(n), ctx) \
+		SPI_DMA_CHANNEL(n, rx) \
+		SPI_DMA_CHANNEL(n, tx) \
```

Comment:
Prefix your macros with `SPI_SILABS_EUSART`. Otherwise, the reader could think `SPI_DMA_CHANNEL()` is provided by the SPI framework.

BTW, `SPI_SILABS_EUSART` is probably too long. I believe the name of the driver (`EUSART`) would be sufficient. However, this should be done in an atomic way, probably in a separate PR.